x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/index.tsx

```diff
@@ -679,6 +682,15 @@ export const EndpointList = () => {
     );
   }, [showTransformFailedCallout, closeTransformFailedCallout, transformFailedCalloutDescription]);
 
+  if (!canReadEndpointList || (canReadEndpointList && !canAccessFleet)) {
+    return (
+      <NoPrivileges
+        pageName={SecurityPageName.endpoints}
+        documentationUrl={services?.docLinks.links.securitySolution.privileges}
+      />
+    );
+  }
```

Comment:
I don't think you need this here. the display of `NoPrivileges` is handled at the Route level in 

https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/management/pages/index.tsx#L99-L103

-----

Also,
Just FYI: the `canAccessFleet` permission will be needed, but not to drive if the page can be accessed. It will be used when you work on the other ticket around the List and Details action menu to control if the menu options around the jumps to Fleet are displayed or not.

Reply:
You're right that the PrivilegedRoute component does handle most of the NoPrivileges display. It currently doesn't catch the fleet access privilege, but that is simple enough to change. According to the ticket, it looks like we still want to restrict the onboarding page if the user does not also have fleet access.

Reply:
Yeah, but that should not prevent the user from accessing the Endpoint list. Fleet:all should not be a requirement for a user to see the entire page.

The Fleet permission will only be used (i suspect) to determine which empty state to show (onboarding or "no items") and the menu options under the action menu that takes the user to the Agent details in fleet.